xmrstak/backend/iBackend.hpp

```diff
@@ -38,7 +38,7 @@ namespace xmrstak
 		// Limited version of the nonce calc above
 		inline uint32_t calc_nicehash_nonce(uint32_t start, uint32_t resume)
 		{
-			return start | ( calc_start_nonce(resume) >> 8u );
+			return start | ( calc_start_nonce(resume) >> 7u );
```

Comment:
Why 7 here? If you want to have the top 8 bits for nicehash and 24 for us, this should be 8.

Reply:
Nicehash blocks the highest 8bit within the 32bit nonce. The `calc_start_nonce` function sets for the thread one the bit 31, by shifting the 31th activated bit 7 places right we end with an activated bit in place 23. This means we use bit [0;23] for the nonce.

Reply:
Still not quite getting it here, let's talk numbers. Is calc_start_noce using everything up to the MSB? So for example 0xFFFF0000 - or does it leave it blank - 0x7FFF0000?

Reply:
Just to give you food for thought:
0xAC000000 | (0xFFFF0000 >> 7) = 0xADFFFE00

Reply:
I fixed it, thx @fireice-uk

Reply:
Yes 7-bit shifting make about 50% share rejected (on cpu it's duplicated share, on gpu it's is wrong nonce).

Try rebuild with 8-bit shifting value, everything seem good now.

Let's it run a bit longer to check do we have out of nonce problem with high hashrate.